lib/puppet/property.rb

```diff
@@ -402,6 +402,7 @@ def property_matches?(current, desired)
   # @param value [Object] the value to format as a string
   # @return [String] a pretty printing string
   def is_to_s(value)
+    value = value[0] if value.is_a?(Array) && value.size == 1
```

Comment:
I don't think we want to do this, otherwise it makes properties sometimes look like a String and sometimes an Array. For example, with this PR:

```
Notice: /Stage[main]/Main/User[foo]/groups: groups changed  to 'ntp'
```

In Puppet 4.10.1:

```
Notice: /Stage[main]/Main/User[foo]/groups: groups changed '' to ['wheel']
```

Since the `groups` property is multi-valued, we should always present it that way.